src/components/cards/productCard.js

```diff
@@ -0,0 +1,53 @@
+import './index.css'
+import { useState } from 'react'
+import {useCart,useNavigate} from '../../customHooks'
+import axios from 'axios'
+const ProductCard=({productObj})=>{
+  const{price,imageUrl,title,description,productRating,readyToDeliver}=productObj
+
+  const {dispatchCart,wishlistCount,cartCount}=useCart()
+  
+  const navigate=useNavigate()
+    const [inCart,setButtonText]=useState(false)
+
+    const addItemToCart=async(product)=>{
+      try{
+        console.log(product,"entered intotry")
+        const response=await axios.post('/api/user/cart',{product},{
+          headers:{
+            authorization:"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJfaWQiOiI3MTUxN2NmOS05MTg4LTRlNGYtOWM1MS0xMzMxZWE1ZThkZmQiLCJlbWFpbCI6ImFkYXJzaGJhbGlrYUBnbWFpbC5jb20ifQ.WTkYnS_dAUXq8sBn-GKoX0BC6ZJKNpL8Q_CNUzlebJI"
```

Comment:
you should not pass these values like this in a hardcoded manner instead you can access that from a variable called token or something like that.